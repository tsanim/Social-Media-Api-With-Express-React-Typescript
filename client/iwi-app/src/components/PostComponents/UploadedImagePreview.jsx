import React from 'react';
import PropTypes from 'prop-types';

function UploadedImagePreview(props) {
    return (
        <div className="uploadedImgPreview">
            {
                props.imagePreviewUrl !== '' ? <React.Fragment>
                    <img id="previewImg" src={props.imagePreviewUrl} alt="uploadedImg" />
                    <div onClick={props.handlePreviewImgClose} className="overlay"><span>x</span></div>
                </React.Fragment> : null
            }
        </div>
    )
}

UploadedImagePreview.propTypes = {
    imagePreviewUrl: PropTypes.string,
    handlePreviewImgClose: PropTypes.func
}

export default UploadedImagePreview;