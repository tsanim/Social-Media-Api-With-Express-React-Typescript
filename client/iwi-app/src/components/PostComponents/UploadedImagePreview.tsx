import React from 'react';
import PropTypes from 'prop-types';
import UploadedImagePreviewProps from '../../interfaces/Components/PostComponents/UploadedImagePreviewProps.interface';

function UploadedImagePreview(props: UploadedImagePreviewProps) {
    return (
        <div className="uploadedImgPreview">
            {
                props.imagePreviewUrl !== '' ? <React.Fragment>
                    <img id="previewImg" src={props.imagePreviewUrl} alt="uploadedImg" />
                    <div onClick={props.handlePreviewImgClose} className="overlay"><span>x</span></div>
                </React.Fragment> : null
            }
        </div>
    )
}

UploadedImagePreview.propTypes = {
    imagePreviewUrl: PropTypes.string,
    handlePreviewImgClose: PropTypes.func
}

export default UploadedImagePreview;