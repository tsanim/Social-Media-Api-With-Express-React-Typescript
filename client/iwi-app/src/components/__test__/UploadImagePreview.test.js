import React from 'react';
import UploadedImagePreview from '../PostComponents/UploadedImagePreview';

import { shallow } from 'enzyme';

function uploadedImagePreviewComponentSetup() {
    const props = {
        imagePreviewUrl: '123',
        handlePreviewImgClose: jest.fn()
    }

    const enzymeWrapper = shallow(<UploadedImagePreview {...props} />)

    return {
        props,
        enzymeWrapper
    }
}

describe('UploadedImagePreview component', () => {
    it('should render initial layout', () => {
        const { enzymeWrapper } = uploadedImagePreviewComponentSetup();

        expect(enzymeWrapper.getElements()).toMatchSnapshot();
    });

    it('should render div tag with class "uploadedImgPreview"', () => {
        const { enzymeWrapper } = uploadedImagePreviewComponentSetup();

        expect(enzymeWrapper.find('div.uploadedImgPreview')).toHaveLength(1);
    });

    it('should render img tag with id "previewImg" when imagePreviewUrl prop is not empty string', () => {
        const { enzymeWrapper } = uploadedImagePreviewComponentSetup();

        expect(enzymeWrapper.find('img#previewImg')).toHaveLength(1);
    });

    it('should render a div tag with class "overlay"  when imagePreviewUrl prop is not empty string', () => {
        const { enzymeWrapper } = uploadedImagePreviewComponentSetup();

        expect(enzymeWrapper.find('div.overlay')).toHaveLength(1);
    });
});
