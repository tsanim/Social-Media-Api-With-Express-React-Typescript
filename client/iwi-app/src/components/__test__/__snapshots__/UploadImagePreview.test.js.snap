// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadedImagePreview component should render initial layout 1`] = `
Array [
  <div
    className="uploadedImgPreview"
  >
    <React.Fragment>
      <img
        alt="uploadedImg"
        id="previewImg"
        src="123"
      />
      <div
        className="overlay"
        onClick={[MockFunction]}
      >
        <span>
          x
        </span>
      </div>
    </React.Fragment>
  </div>,
]
`;
